### PR TITLE
Feature/trips/viewer

### LINKED ISSUE #35: View selected trip on map
Selecting a trip on the Stop viewer opens another fragment that displays the trip on the map

### PR DESCRIPTION
**Pull request by dellisd**
# Issues
Closes #35 

# Summary
Clicking on a trip in the StopFragment will open a map viewer for all of the trips of the selected route.

# Screenshots

### CODE REVIEW

File: app/src/main/java/ca/llamabagel/transpo/ui/trips/TripsFragment.kt [hunk 1]
```diff
@@ -22,6 +49,83 @@ class TripsFragment : Fragment() {
...
+                .map { trip -> Point.fromLngLat(trip.tripUiModel.trip.longitude!!, trip.tripUiModel.trip.latitude!!) }
+                .toList()
+
+            val geometry = Feature.fromGeometry(MultiPoint.fromLngLats(points))
+            if (points.size > 1) {
+                val lngLatBounds =
+                    LatLngBounds.Builder().includes(points.map { LatLng(it.latitude(), it.longitude()) }).build()
+                map.moveCamera(CameraUpdateFactory.newLatLngBounds(lngLatBounds, 32))
+            } else if (points.isNotEmpty()) {
+                map.moveCamera(CameraUpdateFactory.newLatLngZoom(LatLng(points[0].latitude(), points[0].longitude()), 15.0))
```
Comment (GustavoSanMartin) [inline app/src/main/java/ca/llamabagel/transpo/ui/trips/TripsFragment.kt]: Exceeds 80 char limit. Also we shouldn't have magic numbers

File: app/src/main/java/ca/llamabagel/transpo/ui/trips/TripsViewModel.kt [hunk 2]
```diff
@@ -32,12 +42,50 @@ class TripsViewModel @Inject constructor(private val tripsRepository: TripsRepos
...
+            is Result.Success -> {
+                _displayData.value = copy.data.routes.flatMap { route ->
+                    route.trips.map { trip -> TripUiModel(route, trip) }
+                }
+                    .sortedBy { it.trip.adjustedScheduleTime }
+                    .map(::TripItem)
+                updateViewerData()
+            }
+            is Result.Error -> {
+                // TODO: Handle errors
```
Comment (GustavoSanMartin) [inline app/src/main/java/ca/llamabagel/transpo/ui/trips/TripsViewModel.kt]: This TODO intentional?

File: app/src/main/java/ca/llamabagel/transpo/ui/trips/TripUiModel.kt [hunk 3]
```diff
@@ -8,25 +8,10 @@ import ca.llamabagel.transpo.models.trips.Route
...
-    val routeType: String,
-    val destination: String,
-    val adjustedScheduleTime: Int,
-    val adjustmentAge: Float,
-    val lastTripOfSchedule: Boolean,
-    val busType: String,
-    val hasBikeRack: Boolean
+    val route: Route,
+    val trip: Trip,
+    val selected: Boolean = false
```
Comment (GustavoSanMartin) [inline app/src/main/java/ca/llamabagel/transpo/ui/trips/TripUiModel.kt]: You can fit this on a single line

File: app/src/main/java/ca/llamabagel/transpo/ui/trips/TripsViewModel.kt (hunk 2 above)
Comment (dellisd) [reply app/src/main/java/ca/llamabagel/transpo/ui/trips/TripsViewModel.kt]: For now yeah, not sure how we're going to handle error states.

File: app/src/main/java/ca/llamabagel/transpo/ui/trips/TripsFragment.kt (hunk 1 above)
Comment (dellisd) [reply app/src/main/java/ca/llamabagel/transpo/ui/trips/TripsFragment.kt]: tEmpOrARy

...but ok

File: app/src/main/java/ca/llamabagel/transpo/ui/search/viewholders/SearchStopViewHolder.kt [hunk 4]
```diff
@@ -15,5 +15,4 @@ class SearchStopViewHolder(private val binding: SearchStopBinding) : RecyclerVie
     fun bind(stop: SearchResult.StopItem) {
         binding.stop = stop
     }
-
```
[the remark this answers is not in the document]
Comment (dellisd) [reply app/src/main/java/ca/llamabagel/transpo/ui/search/viewholders/SearchStopViewHolder.kt]: What does that mean?

[the remark this answers is not in the document]
Comment (GustavoSanMartin) [reply app/src/main/java/ca/llamabagel/transpo/ui/search/viewholders/SearchStopViewHolder.kt]: removing the line doesn't contribute to anything and is the only change in the file. (so don't change it)

[the remark this answers is not in the document]
Comment (dellisd) [reply app/src/main/java/ca/llamabagel/transpo/ui/search/viewholders/SearchStopViewHolder.kt]: It was reported by ktlint

[the remark this answers is not in the document]
Comment (GustavoSanMartin) [reply app/src/main/java/ca/llamabagel/transpo/ui/search/viewholders/SearchStopViewHolder.kt]: ah ok

**Pull request merged**